server/controllers/petProfile.js

```diff
@@ -0,0 +1,74 @@
+const PetProfile = require("../models/PetProfile");
+const asyncHandler = require("express-async-handler");
+
+exports.createPetProfile = asyncHandler(async (req, res, next) => {
+  const { petName, ownerID, petType, petAge, petStatus } = req.body;
+  if (!petName) {
+    res.status(404);
+    throw new Error("No pet name provided");
+  }
+  if (!ownerID) {
+    res.status(404);
```

Comment:
These aren't the right status codes here, the right status code would be 400 HTTP code.

Reply:
Q: If we have middleware validation why do we need to do this actually? We could juse verify no errors occured in the validation result as opposed to doing this here?

Reply:
OK- makes sense I will remove